src/Features/CSharp/Portable/GenerateDefaultConstructors/CSharpGenerateDefaultConstructorsService.cs

```diff
@@ -45,19 +45,14 @@ public CSharpGenerateDefaultConstructorsService()
 
             var syntaxTree = semanticDocument.SyntaxTree;
             var node = semanticDocument.Root.FindToken(textSpan.Start).GetAncestor<TypeSyntax>();
-            if (node != null)
+            if (node is { Parent: BaseTypeSyntax { Parent: BaseListSyntax baseList } })
             {
-                if (node.Parent is BaseTypeSyntax && node.Parent.Parent is BaseListSyntax baseList)
+                if (baseList.Parent is TypeDeclarationSyntax(SyntaxKind.ClassDeclaration or SyntaxKind.RecordDeclaration) parentTypeDecl &&
+                    baseList.Types.Count > 0 &&
+                    baseList.Types[0].Type == node)
```

Comment:
Disappointed you didn't keep the trend going
```suggestion
                    baseList.Types is [{Type: var nodeType}, ..] &&
                    nodeType == node)
```

Reply:
so in cases like these i didn't bother if it would still be multiple checks :)  but i def am ok if someone wants to code this in either form.

Reply:
Fixed up.